src/intersect-polygon/intersect-polygon.module.js

```diff
@@ -53,49 +65,65 @@ const intersectPolygon = (georaster, geometry, perPixelFunction, { debug_level =
 
     const sample_bboxes = usedPercentage > 0.01 ? [combined_geometry_bbox] : geometry_bboxes;
 
-    // get samples for each geometry bbox
-    samples = sample_bboxes.map(sample_bbox => {
+    const sample_image_bboxes = sample_bboxes.map(sample_bbox => {
       const [xmin, ymin, xmax, ymax] = sample_bbox;
 
-      // snap whole geometry bounding box to georaster grid system
-      const snapResult = snap({
+      const snap_params = {
         bbox: [xmin.toString(), ymin.toString(), xmax.toString(), ymax.toString()],
         debug: false,
         origin: [georaster.xmin.toString(), georaster.ymax.toString()],
         overflow: false,
+        padding: ["1", "1"], // add a little padding in case the geometry is smaller than half a pixel
```

Comment:
Will this always apply 1 pixel padding then?  It makes sense

Reply:
yup, it'll add 1 pixel padding all around, so technically 2 new pixels along the horizontal axis and 2 along the vertical axis.  if the geometry is really small and the original width is 0 pixels, the new width will be 2 pixels.

it won't solve the issue of trying to grab values when a geometry is a lot smaller than a pixel size, but my thinking is that it'll be one piece in that solution